torch/distributed/fsdp/_init_utils.py

```diff
@@ -499,11 +505,13 @@ def _init_param_handle_from_params(
     assert handle not in state._handles
     state.params.append(handle.flat_param)
     state._handles.append(handle)
-    state._comm_module_to_handles[handle._comm_module].append(handle)
-    num_comm_module_handles = len(state._comm_module_to_handles[handle._comm_module])
-    assert num_comm_module_handles == 1, (
+    state._fully_sharded_module_to_handles[handle._fully_sharded_module].append(handle)
+    num_fully_sharded_module_handles = len(
+        state._fully_sharded_module_to_handles[handle._fully_sharded_module]
+    )
+    assert num_fully_sharded_module_handles == 1, (
         "The current design assumes a module manages at most one "
-        f"`FlatParamHandle` but got {num_comm_module_handles}"
+        f"`FlatParamHandle` but got {num_fully_sharded_module_handles}"
```

Comment:
unrelated to this PR, but are there any plans / use cases to relax this assumption?

Reply:
Nothing in the near future. From what I can tell, it would only be motivated by some fine tuning use cases. This might be something for post-stable release.